### PR TITLE
posix: add SetHwFlowCtrl()

### PR DESCRIPTION
**Pull request by liamstask**
tested on OS X and Linux

### CODE REVIEW

File: term_posix.go [hunk 1]
```diff
@@ -120,6 +120,20 @@ func (t *Term) SetRTS(v bool) error {
 	}
 }
 
+// SetHwFlowCtrl specifies whether hardware flow control is enabled
+func (t *Term) SetHwFlowCtrl(enabled bool) error {
```
Comment (davecheney) [inline term_posix.go]: I'd like to see this presented as an option, see #13. 

I'd like to see a friendlier name, like FlowControl(term.Hardware|term.Software) also as an option, etc

**Comment by davecheney** [comment]
LGTM with comments. Can you please add an example for how this is used.

**Comment by liamstask** [comment]
in terms of naming, maybe just `SetFlowCtrl(v bool)` would work?

in terms of the functionality provided, the hw flow control is either enabled or disabled, so i think a `term.Software` option would be a bit misleading (since that would imply there's some sw flow control, which is not the case).

thoughts?

**Comment by liamstask** [comment]
pushed some proposed changes, happy to modify them further.

also, i suspect this may cause merge conflicts with #13 - if that gets merged first, I'll plan on rebasing this PR to fix it up.

**Comment by davecheney** [comment]
Aren't there more than two types of flow control? None, xon/xoff/hardware.
Are these all exposed via termios?

**Comment by liamstask** [comment]
gotcha - i'll put together a patch that adds support for XON/XOFF sw flow control as well.

in that case, do you figure it's worth defining a flow control type that has 3 possible values, none/hardware/software? (sorry if this is what you already essentially suggested above, just want to make sure i understand).

**Comment by davecheney** [comment]
Lets see how far

const (
    NONE = iota
    XONXOFF
    HARDWARE
)

func SetFlowCntl(kind int) ...

Goes, if it needs more ceremony it can always be added later.

**Comment by liamstask** [comment]
updated, let me know what you think

File: term.go [hunk 2]
```diff
@@ -12,6 +12,12 @@ import (
 	"github.com/pkg/term/termios"
 )
 
+const (
+	NONE     = iota // flow control off
```
Comment (davecheney) [inline term.go]: If you move the comments above the declaration, they'll show up in godoc

// Flow control is disabled
NONE = iota

// Software flow control
XONXOFF

**Comment by davecheney** [comment]
LGTM with one little tweak. Do you want to finish up this PR and i'll merge it for you.

Thanks

Dave

File: term.go (hunk 2 above)
Comment (liamstask) [inline term.go]: huh - i see them show up in godoc locally as is. do they show up anywhere other than the `Constants` section on the package page?

happy to move them either way

Comment (davecheney) [inline term.go]: Hmm, let me check my godoc.

**Comment by liamstask** [comment]
going to rebase on master to resolve conflicts

**Comment by liamstask** [comment]
ok all set, standing by on whether the comments ultimately need to be moved.

**Comment by davecheney** [comment]
LGTM. We can bikeshed the comment another day, it doesn't need to block you.

**Comment by liamstask** [comment]
ok - thx!

**Pull request merged**